src/CoreWCF.Primitives/src/CoreWCF/Description/TypeLoader.cs

```diff
@@ -204,6 +204,28 @@ private ContractDescription LoadRequestChannelContractDescription()
             return contract;
         }
 
+        //This needs proper review. 
+        private void AddAuthorizeOperations(ContractDescription contractDesc, bool implIsCallback, ContractReflectionInfo reflectionInfo)
+        {
+
+            for(int i=0; i< contractDesc.Operations.Count; i++)
+            {
+                OperationDescription opDesc = contractDesc.Operations[i];
+                Type targetIface = implIsCallback ? opDesc.DeclaringContract.CallbackContractType : opDesc.DeclaringContract.ContractType;
+                ApplyServiceInheritance(
+                    opDesc.AuthorizeOperation,
+                    delegate (Type currentType, KeyedByTypeCollection<IAuthorizeOperation> behaviors)
+                    {
```

Comment:
This could benefit from using a local function. See [here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions) for details.

Reply:
need little more explanation on how ?